cpp/src/io/parquet/chunk_dict.cu

```diff
@@ -192,31 +194,30 @@ __global__ void __launch_bounds__(block_size, 1)
   auto map =
     map_type::device_view(chunk.dict_map_slots, chunk.dict_map_size, KEY_SENTINEL, VALUE_SENTINEL);
 
-  __shared__ size_type counter;
-  if (t == 0) counter = 0;
+  __shared__ cuda::atomic<size_type, cuda::thread_scope_device> counter;
```

Comment:
Since it's shared, only threads in the same CTA can access it, which means we can use the narrower "block" scope. 
```suggestion
  __shared__ cuda::atomic<size_type, cuda::thread_scope_block> counter;
```